core/store/src/trie/iterator.rs

```diff
@@ -130,6 +130,15 @@ impl<'a> TrieIterator<'a> {
         result
     }
 
+    fn has_value(&self) -> bool {
+        let b = self.trail.last().expect("Trail finished.");
```

Comment:
is this function guaranteed to not be called when trail is empty?

Reply:
No, but AFAIU this case corresponds to the end of trie, so it can be handled trivially.
@abacabadabacaba is that correct?